nilearn/reporting/tests/test_reporting.py

```diff
@@ -2,20 +2,17 @@
 import pytest
 from nibabel import Nifti1Image
 
+from nilearn._utils.helpers import is_matplotlib_installed
 from nilearn.image import get_data
 
 # Set backend to avoid DISPLAY problems
-from nilearn.plotting import _set_mpl_backend
+from nilearn.plotting import _set_mpl_backend  # noqa
```

Comment:
@Remi-Gau I think there is still a problem with this line if I am not missing something. I am not sure if this line is necessary.

And a second problem is with the import below:

```
from nilearn.reporting import get_clusters_table
from nilearn.reporting.get_clusters_table import (
    _cluster_nearest_neighbor,
    _local_max,
)
```
These two imports causes in  `nilearn.reporting.__init__` file to run the below line

`from nilearn.reporting.glm_reporter import make_glm_report`  

which causes `matplotlib` to be imported and when it is not installed, it does not skip the tests and gives error.


I think we cannot detect it because  CI tests do not have a configuration where `matplotlib` is not installed. 

I propose to create an issue on that.

Reply:
hmm...

we also run tests with none of the plotting dependencies:

github workflow
https://github.com/nilearn/nilearn/blob/dfe904a95f54ea443e7a9cba60da9d9f3b76bd13/.github/workflows/test_with_tox.yml#L37-L40

calls tox env
https://github.com/nilearn/nilearn/blob/dfe904a95f54ea443e7a9cba60da9d9f3b76bd13/tox.ini#L117-L124

with minimum dependencies and no plotting
https://github.com/nilearn/nilearn/blob/dfe904a95f54ea443e7a9cba60da9d9f3b76bd13/tox.ini#L16-L25

apparently it ran without failure:
https://github.com/nilearn/nilearn/actions/runs/11481708050/job/31953042366?pr=4675

You should be able to check things as they are ran in CI with the following in python 3.9.

```bash
pip install tox
tox run -e test_min -- nilearn/reporting/tests/test_reporting.py
```

More on tox here: https://nilearn.github.io/dev/maintenance.html#using-tox

I may defo have overlooked something, so please investigate !

Reply:
> with minimum dependencies and no plotting
> 
> https://github.com/nilearn/nilearn/blob/dfe904a95f54ea443e7a9cba60da9d9f3b76bd13/tox.ini#L16-L25

Ok, I hadn't check this one.

> apparently it ran without failure: https://github.com/nilearn/nilearn/actions/runs/11481708050/job/31953042366?pr=4675
> 

However, for the above run, the tests under `nilearn/reporting` are not run. Is there a condition somewhere that excludes this package when `matplotlib` is not installed?

[For this test](https://github.com/nilearn/nilearn/actions/runs/11481708050/job/31953042849?pr=4675) when matplotlib is installed, tests for `nilearn/reporting` are run.

> You should be able to check things as they are ran in CI with the following in python 3.9.
> 
> ```shell
> pip install tox
> tox run -e test_min -- nilearn/reporting/tests/test_reporting.py
> ```

When I run this, it gives error on my local.

Reply:
> Is there a condition somewhere that excludes this package when `matplotlib` is not installed?

I think this part of the conftest.py is what does this.

https://github.com/nilearn/nilearn/blob/6a7a8078ef4a26e2d49203eb599cb0e395808cfe/nilearn/conftest.py#L27-L38

Reply:
> When I run this, it gives error on my local.

OK
Then maybe we should not have those global collect ignore in the conftest.py otherwise it 'hides' some potential failures. What do you think?

Reply:
Yes, I agree.

Reply:
Do you want to give this a go at fixing this?